browser/custom_app_icons/android/BUILD.gn

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) 2025 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import("//build/config/android/rules.gni")
+import("//build/config/locales.gni")
+import("//chrome/common/features.gni")
+import("//tools/grit/grit_rule.gni")
+
+android_library("java") {
+  sources = [
+    "java/src/org/chromium/brave/browser/custom_app_icons/CustomAppIconsEnum.java",
+    "java/src/org/chromium/brave/browser/custom_app_icons/CustomAppIconsManager.java",
+    "java/src/org/chromium/brave/browser/custom_app_icons/settings/CustomAppIconsAdapter.java",
+    "java/src/org/chromium/brave/browser/custom_app_icons/settings/CustomAppIconsFragment.java",
+    "java/src/org/chromium/brave/browser/custom_app_icons/settings/CustomAppIconsListener.java",
+  ]
+  deps = [
+    ":java_resources",
+    "//base:tasks_java",
+    "//brave/browser/utils/android:java",
+    "//chrome/android:chrome_base_module_resources",  # adding dep for
```

Comment:
why does this dep reqiure a comment here? We typically do not add comments explaining each dep, we just add what `gn_check` requires unless it's some strange situation where we're adding a dep that doesn't seem like it should be needed

Reply:
also why is it formatted so strangely? I'm not sure where the 120 character limit is here, but a comment like this would normally go above anyway

Reply:
resolved here : https://github.com/brave/brave-core/pull/28287/commits/73f5122ac47c6c2799580c1fab6e7e944f1d1203